File: src/main/java/lkm/starterproject/accounting/entity/register/Card.java
package lkm.starterproject.accounting.entity.register;

import jakarta.persistence.*;
import lkm.starterproject.accounting.constants.UseStatus;
import lkm.starterproject.accounting.entity.basic.Address;
import lkm.starterproject.accounting.entity.company.Company;
import lkm.starterproject.auth.entity.BaseEntity;
import lombok.*;

import java.time.LocalDate;

@Getter @Setter
@Builder @NoArgsConstructor @AllArgsConstructor
@Entity @Table(name = "card")
public class Card extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "card_id")
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "company_id")
    private Company company;

    private Long code;  //개인 코드

    @Column(nullable = false)
    private String name;    //카드(사)명

    @Column(nullable = false)
    private String cardNumber;  //카드번호

    private String note;    //비고

    private String division;    //거래구분

    @Enumerated(EnumType.STRING)
    private UseStatus useStatus;    //사용구분

    @ManyToOne
    @JoinColumn(name = "finance_id")
    private Finance finance;

    @Column(length = 13)
    private String phone;   //전화번호

    @Column(length = 13)
    private String fax;     //팩스번호

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "address_id")
    private Address address;    //주소

    @Column(length = 12)
    private String registrationNumber;  //사업자등록번호

    private String homePage;        //홈페이지주소

    private LocalDate contractStartDate;    //계약시작일

    private LocalDate contractEndDate;       //계약종료일

    private Double commission;      //수수료

    private LocalDate expirationStartDate;  //유효기간 시작일

    private LocalDate expirationEndDate;     //유효기간 종료일

    private String chargeName;  //담당자명

    private LocalDate accountDay;   //결제일

}

File: vue/types/accounting/card.ts
import type {Company} from "~/types/accounting/company";
import type {Address} from "~/types/accounting/basicdata/address";
import type {Finance} from "~/types/accounting/finance";

export interface Card {
    id: number;
    company: Company;
    name: string;    //카드사 명
    cardNumber: string;  //카드(가맹점)번호
    note: string;    //비고
    division: string; //매입, 매출 구분
    useStatus: string;    //사용구분
    finance: Finance;   //결제/입금 계좌
    phone: string;   //전화번호
    fax: string;     //팩스번호
    address: Address;    //주소
    registrationNumber: string;  //등록번호
    homePage: string;    //홈페이지주소
    /*매출*/
    contractStartDate: Date; //계약기간 시작일
    contractEndDate: Date; //계약기간 종료일
    commission: number;    //수수료
    /*매입*/
    expirationStartDate: Date;  //유효기간 시작일
    expirationEndDate: Date; //유효기간 종료일
    chargeName: String; //담당자이름
    accountDay: Date; //결제일
}
